### ISSUE TITLE
amplify mock: Failed to fetch

### ISSUE THREAD

**Issue by cyrfer** [aws-amplify/amplify-cli#4625]
**Describe the bug**
This seems closely related to another bug report, but I'm going to include more details that I think are interesting.
https://github.com/aws-amplify/amplify-cli/issues/4240

To support a big refactor, I wanted to leverage the local server abilities of `amplify mock`. I attempted to run my code that works in the cloud, locally, and noticed I had some problems locally.

When running my GraphQL API locally using `amplify mock` with Lambda responsible for the response, GraphiQL hangs and eventually I see this response:
```json
{
  "message": "Failed to fetch",
  "stack": "TypeError: Failed to fetch"
}
```

My schema looks like this (omitting unnecessary details):
```graphql
type TitleTalentRoleInfo {
  creditSequence: Int
  character: String
  role: String
}

type TitleTalent {
  label: String
  talentData: [TitleTalentRoleInfo!]
}

type LivesearchTitle {
  title: String
  type: String
  talent: [TitleTalent!]
}

type LocalizedTitleSet {
  title: LivesearchTitle!
}

type Query {
  searchTitle(phrase: String!): [LocalizedTitleSet!]! @function(name: "l12nAPI-${env}")
}
```

The problem is not in my lambda. I am using `console.log` to see the Lambda behaves as expected by reading what is printed by my lambda in the `amplify mock` terminal.

In some cases, I have been able to avoid the problem by changing my Lambda to redundantly recreate the response data using `Array.map`. Perhaps something about nested arrays is the problem. I'm guessing some of my data attributes are `undefined` or `null` and that makes problems for the local server. Maybe using `Array.map` ensures a response that fits with the schema? Again, the original implementation without the `Array.map` workaround has been working well in the cloud.

For example, this use of `talent` in my lambda response makes the local server hang, with the eventual undesired response,
```javascript
  return shape.map(s => (
    {
      title: {
        title: s.title.title,
        type: s.title.type,
        talent: (s.title.talent || []).map(t => t),
      }
    }
  ))
```

While this use of `talent` in my lambda response allows the local server to succeed:
```javascript
  return shape.map(s => (
    {
      title: {
        title: s.title.title,
        type: s.title.type,
        talent: (s.title.talent || []).map(t => ({
          label: t.label,
          talentData: (t.talentData || []).map(td => ({
            creditSequence: td.creditSequence,
            character: td.character,
            role: td.role,
          }))
        })),
```

I use this request in GraphiQL to test the code above:
```graphql
query SearchTitles($phrase:String!) {
  searchTitle(phrase:$phrase) {
    title {
      title
      type
      talent {
        label
      }
    }
  }
}
```

**Amplify CLI Version**
4.21.0

**To Reproduce**
1. run `amplify api add`
2. run `amplify function add`
3. modify schema.graphql Query field to use a function returning arrays of objects with nested arrays.
4. print the lambda response in your function to be sure the lambda is working.
5. run `amplify mock`
6. make a request in GraphiQL

**Expected behavior**
I expect my lambda returning this payload, including `talent`, to work...
```javascript
  return shape.map(s => (
    {
      title: {
        title: s.title.title,
        type: s.title.type,
        talent: (s.title.talent || []).map(t => t),
      }
    }
  ))
```

Well, I expect this lambda implementation to work also!
```javascript
    return shape
```

...with this request (including `talent`):
```graphql
query SearchTitles($phrase:String!) {
  searchTitle(phrase:$phrase) {
    title {
      title
      type
      talent {
        label
      }
    }
  }
}
```

**Desktop (please complete the following information):**
 - OS: *Mac High Sierra*
 - Node Version: *v12.16.2*

**Additional context**
My schema is much more elaborate but I wanted to keep the size of this post smallish.

**Comment by akshbhu** [comment]
Hi @cyrfer 

Thanks for reporting this.Could you provide your lambda function so that we can try reproduce this issue?

**Comment by cyrfer** [comment]
I submitted the code sample and schema via aws support.

**Comment by akshbhu** [comment]
Hi @cyrfer 

I have reproduced the bug. It seems there is a limit on sending large data in mock.

**Comment by cyrfer** [comment]
Yes, in my attempts to recreate the result object, I must have been filtering out some amount of data, going below the (undocumented) limit threshold.

The sample provided through aws support was just large enough to show the problem. Simply reducing the payload size avoids the limit threshold.

**Comment by akshbhu** [comment]
Hi cryfer

I have changed the logic to avoid the limit on the payload size and will submit a PR for this fix.

**Comment by cyrfer** [comment]
@akshbhu what is the PR?

**Comment by akshbhu** [comment]
The fix is included in amplify-cli version 4.26.0. Let me know if you anymore issues , comment on this and I will open the issue again.
